api/nakadi-event-bus-api.yaml

```diff
@@ -1250,6 +1250,8 @@ definitions:
 
           To have a generic, undefined schema it is possible to define the schema as `"schema":
           "{\"additionalProperties\": true}"`.
+
+          If type of schema is JSON_SCHEMA then schema can be passed as json object for convenience.
```

Comment:
Object level schema would be awesome. I'd avoid overloading the type of the `schema` field as that's going to complicate readers, even though it can be done by stringiness checking of the data when it's received. Defining a `json_schema_object` enum option and a new `schema_object` field would provide a clean form of extension - if the enum is `json_schema_object` look in the `schema_object` field, no jackson modules needed - it's less magical :) Also for the json schema case (and avro if that gets supported), it'll be easy for the server to populate either field from the other. 

```yaml
  EventTypeSchema:
    properties:
      type:
        type: string
        enum:
          - json_schema
          - json_schema_object
      schema:
        type: string
      schema_object:
        type: object
    required:
      - type
```

Fwiw, today's string schema field is the most generic  option as it can support non-JSON schema (eg swagger yaml or protobufs), so it's probably worth keeping around with schema_object as a special consideration for the current common case (json formatted schema definitions)